### ISSUE TITLE
type(*) not working

### ISSUE THREAD

**Issue by jkd2016** [lfortran/lfortran#2415]
When compiling

```
type(*), intent(in) :: buffer(..)
```

LFortran reports

```
semantic error: Type must have a name
  --> mpi_stub.f90:65:1
   |
65 | type(*), intent(in) :: buffer(..)
   | ^^^
```
This is a Fortran 2018 feature and is needed to interface to the MPI subroutines.

(I hope you don't mind me reporting these problems but I'm currently attempting to compile our electronic structure code [Elk](https://elk.sourceforge.io/) with LFortran to see how far I can get.)

**Comment by certik** [comment]
Yes, we don't support that yet, and I think `a(..)` doesn't work either.

@jkd2016 Not at all, thank you for all the reports! Please report as many bugs as you can. We will definitely get to compiling Elk, but I think it's a bit early, we still need to compile a few simpler codes first. But definitely report all the bugs that you can, we'll fix them.
